Backpack-SwiftUI/Select/Classes/BPKSelect.swift

```diff
@@ -0,0 +1,170 @@
+        options: [String],
+        selectedIndex: Int,
+        onSelectionChange: @escaping (Int) -> Void
+    ) {
+        self.options = options
+        self.selectedIndex = selectedIndex
+        self.title = placeholder
+        self.onSelectionChange = onSelectionChange
+    }
+    
+    public var body: some View {
+        VStack {
+            
+            Picker(title, selection: $selectedIndex) {
+                ForEach(indexIdentifiableOptions) { t in
+                    Text(t.string)
+                }
+            }
+            .customPickerStyle(
+                labelText: selectedIndex < options.count ? options[selectedIndex] : "",
+                textColor: state.textColor
+            )
+            .background(.surfaceDefaultColor)
+            .clipShape(RoundedRectangle(cornerRadius: .sm))
+            .disabled(state.isDisabled)
+            .outline(state.borderColor, cornerRadius: .sm)
+            .onChange(of: selectedIndex) { tag in
+                onSelectionChange(tag)
+            }
+        }
+    }
+    
+    public func inputState(_ state: State) -> BPKSelect {
+        var result = self
+        result.state = state
+        return result
+    }
+}
+
+struct BPKSelect_Previews: PreviewProvider {
+    static var previews: some View {
+        VStack {
+            BPKText("Breakfast Choices", style: .heading1)
+            BPKSelect(
+                placeholder: "Breakfast Choices",
+                options: ["Porridge",
+                          "Eggs",
+                          "Swift UI"],
+                selectedIndex: 99) { index in
+                    
+                }
+            BPKSelect(
+                placeholder: "Empty List",
+                options: [],
+                selectedIndex: 99) { index in
+                    
+                }
+            BPKSelect(
+                placeholder: "Disabled Choices",
+                options: ["This Picker is Disabled",
+                          "You must eat",
+                          "Porridge"],
+                selectedIndex: 99) { index in
+                    
+                }
+            .inputState(.disabled)
+            BPKSelect(
+                placeholder: "Bad Choices",
+                options: ["Eat Metal",
+                          "Or Cement ",
+                          "Maybe some Java?"],
+                selectedIndex: 99) { index in
+                    
+                }.inputState(.error)
+            BPKSelect(
+                placeholder: "Out Of Bounds",
+                options: ["Selected Index Fail"],
+                selectedIndex: 99) { index in
+                    
+                }.inputState(.error)
+        }
+        .padding()
+        .background(.canvasContrastColor)
+        .outline(.surfaceHighlightColor, cornerRadius: .sm)
+    }
+}
+
+extension BPKSelect {
+    private struct IdentifiableString: Identifiable {
+        let string: String
+        let id: Int
+    }
+}
```

Comment:
If you apply above comment, this should not be needed, and can be deleted

Reply:
again, let's discuss, I was fighting with this for a while, and this was the simplest solution I came to that gave the correct API

Reply:
changed in: https://github.com/Skyscanner/backpack-ios/pull/1708/commits/e869ab618cf4f5b4bd63c6bcc874610a21b8d83a